src/quacc/recipes/espresso/phonons.py

```diff
@@ -370,7 +371,7 @@ def _ph_recover_subflow(grid_results: list[RunSchema]) -> RunSchema:
 
     @subflow
     def _grid_phonon_subflow(
-        ph_input_data: dict | None,
+        ph_input_data: UserDict | None,
```

Comment:
Why use `UserDict` instead  of `dict` here?

Reply:
Because ph_input_data is passed to the Namelist class in [line 400](https://github.com/Quantum-Accelerators/quacc/blob/7d4de2b55dbd81bcffe935717e27f028787dff49/src/quacc/recipes/espresso/phonons.py#L398), which inherits from UserDict, I used UserDict. So, the type can be either Namelist or UserDict.

Reply:
Makes sense!